components/view-switcher/view-switcher.js

```diff
@@ -0,0 +1,82 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit';
+
+/**
+ * A segmented button component.
+ */
+class ButtonSegmented extends LitElement {
+
+	static get properties() {
+		return {
+			/**
+			 * ACCESSIBILITY: Label for the segmented button
+			 * @type {string}
+			 */
+			label: { type: String, required: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.container {
+				align-items: center;
+				background-color: var(--d2l-color-gypsum);
+				border-radius: 0.3rem;
+				column-gap: 0.3rem;
+				display: flex;
+				min-height: calc(2rem + 2px);
+			}
+		`;
+	}
+
+	get items() {
+		return this.shadowRoot.querySelector('slot').assignedElements({ flatten: true }).filter(e => e.tagName.toLowerCase() === 'd2l-view-switcher-item-button');
+	}
+
+	render() {
+
+		return html`
+			<div
+				class="container"
+				role="group"
+				aria-label="${this.label}"
+				@d2l-view-switcher-item-button-select=${this.#handleItemSelect}
+				@keydown="${this._handleArrowKeys}">
+				<slot @slotchange="${this.#handleSlotChange}"></slot>
+			</div>
+		`;
+	}
+
+	focus() {
+		const items = this.items;
```

Comment:
Usually for something like this we try to handle the case where the consumer has called `focus` before the shadowDOM has been created, and therefore children distributed to the slot. I don't think we'll be able to use `FocusMixin` per se but we probably should do the same thing. I think there are a couple other examples in core... I will try to find one.